XMonad/Hooks/ScreenCorners.hs

```diff
@@ -108,6 +127,30 @@ createWindowAt SCRight = withDisplay $ \dpy ->
       threshold = 150
    in createWindowAt' (fi w) threshold 1 (fi $ fi h - threshold * 2)
 
+-- "Translate" a ScreenCorner to real (x,y) Positions on a specific monitor
+createWindowAtMonitor :: ScreenCorner -> Int -> Dimension -> X Window
+createWindowAtMonitor corner monitorNumber hotZoneSize = withDisplay $ \dpy -> do
+  screens <- io $ xineramaQueryScreens dpy
+  case screens of
+    Just scrs | monitorNumber < length scrs -> do
+      let XineramaScreenInfo _ x y w h = scrs !! monitorNumber
+          hotZoneSize' = fromIntegral hotZoneSize :: Int
+          x' = fromIntegral x :: Int
+          y' = fromIntegral y :: Int
+          w' = fromIntegral w :: Int
+          h' = fromIntegral h :: Int
+          (xPos, yPos, width, height) = case corner of
+            SCUpperLeft -> (x', y', hotZoneSize', hotZoneSize')
+            SCUpperRight -> (x' + w' - hotZoneSize', y', hotZoneSize', hotZoneSize')
+            SCLowerLeft -> (x', y' + h' - hotZoneSize', hotZoneSize', hotZoneSize')
+            SCLowerRight -> (x' + w' - hotZoneSize', y' + h' - hotZoneSize', hotZoneSize', hotZoneSize')
+            SCTop -> (x' + 150, y', w' - 300, hotZoneSize')
+            SCBottom -> (x' + 150, y' + h' - hotZoneSize', w' - 300, hotZoneSize')
+            SCLeft -> (x', y' + 150, hotZoneSize', h' - 300)
+            SCRight -> (x' + w' - hotZoneSize', y' + 150, hotZoneSize', h' - 300)
+      createWindowAt' (fi xPos) (fi yPos) (fi width) (fi height)
+    _ -> error $ "Invalid monitor number or no screens available for monitorNumber=" ++ show monitorNumber
```

Comment:
I don't like this behavior: basically, if I unplug my monitor, and I have a configuration for two monitors, xmonad will throw an error. I suggest that the rules for non-existent monitors are simply ignored